dev-docs/services/ci/README.md

```diff
@@ -0,0 +1,114 @@
+# The CI Service in Hail Batch
+
+Hail Batch includes a CI service which has three jobs:
+
+- Runs tests against pull requests
+- Merges PRs to the `main` branch
+- Deploys services to the live infrastructure
+
+It does all of these things only if certain conditions are met.
+
+## Deployment
+
+The CI service itself is deployed as a Kubernetes service in the Hail Batch cluster. See 
+[architecture diagram](../Hail%20Batch%20Architectural%20Diagram.png).
+
+As part of its configuration, CI must be configured with an access token allowing it to operate on 
+behalf of a github account called hail-ci-robot.
+
+To make CI work it needs to receive update triggers from the github repository. These are
+configured within the github repository itself.
+
+## Running tests against pull requests
+
+The CI service is constantly monitoring for new PRs in the `hail` repository. 
+
+When a new PR is opened, or an existing PR changes, hooks are triggered which alert the CI service. 
+The CI service will then:
+
+- Check for validity
+  - The author must be a known developer (or else a commit SHA can be individually approved via the CI service UI)
+  - The PR must be against the `main` branch
+- Run tests
+  - The CI service will run tests against the PR
+  - The CI service will report the results of the tests back to GitHub
+
+### Running Tests
+
+The process of running tests goes like:
+
+- CI will generate a new batch and submit it against the production Hail Batch service using its own service credentials
+- The batch contains jobs which will:
+  - Clone the appropriate branch and build a new set of docker images for the updated services.
+  - Deploy the batch suite of k8s services into one of many CI-owned namespaces in the Hail Batch cluster
+    - These namespaces are called 
+  - Run a series of tests against the services
```

Comment:
It also stands up a private instance of the batch services and tests against those!

Reply:
Oh i see you mentioned that below. nice.